tty.go

```diff
@@ -0,0 +1,97 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"net"
+	"os"
+	"os/signal"
+	"path"
+	"syscall"
+
+	"github.com/hyperhq/runv/hypervisor"
+	"github.com/hyperhq/runv/lib/term"
+)
+
+type tty struct {
+	vm           *hypervisor.Vm
+	containerDir string
+	tag          string
+	termFd       uintptr
+	terminal     bool
+}
+
+type ttyWinSize struct {
+	Tag    string
+	Height int
+	Width  int
+}
+
+func newTty(vm *hypervisor.Vm, containerDir string, tag string, termFd uintptr, terminal bool) *tty {
+	return &tty{
+		vm:           vm,
+		containerDir: containerDir,
+		tag:          tag,
+		termFd:       termFd,
+		terminal:     terminal,
+	}
+}
+
+func (tty *tty) resizeTty() {
+	if !tty.terminal {
+		return
+	}
+
+	height, width := getTtySize(tty.termFd)
+	if tty.vm != nil {
+		tty.vm.Tty(tty.tag, height, width)
+		return
+	}
+
+	conn, err := net.Dial("unix", path.Join(tty.containerDir, "runv.sock"))
```

Comment:
resizeTty can be called many time before runv exit, had better to close conn after use it.